SwrveSDKTest/src/test/java/com/swrve/sdk/ConversationActivityTest.java

```diff
@@ -0,0 +1,194 @@
+        intent1.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent1.putExtra("conversation", conversation);
+        conversationActivityController1.withIntent(intent1).create().start().visible();
+        ConversationActivity conversationActivityOrientation1 = conversationActivityController1.get();
+        assertNotNull(conversationActivityOrientation1);
+
+        // toggle orientation
+        int currentOrientation = conversationActivityOrientation1.getResources().getConfiguration().orientation;
+        boolean isPortraitOrUndefined = currentOrientation == Configuration.ORIENTATION_PORTRAIT || currentOrientation == Configuration.ORIENTATION_UNDEFINED;
+        int toOrientation = isPortraitOrUndefined ? Configuration.ORIENTATION_LANDSCAPE : Configuration.ORIENTATION_PORTRAIT;
+        RuntimeEnvironment.application.getResources().getConfiguration().orientation = toOrientation;
+
+        Bundle bundle = new Bundle();
+        conversationActivityController1.saveInstanceState(bundle).pause().stop().destroy();
+
+        ActivityController<ConversationActivity> conversationActivityController2 = Robolectric.buildActivity(ConversationActivity.class);
+        Intent intent2 = new Intent(RuntimeEnvironment.application, ConversationActivity.class);
+        intent2.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
+        intent2.putExtra("conversation", conversation);
+        conversationActivityController2.withIntent(intent2).create(bundle).start().visible().restoreInstanceState(bundle).resume();
+        ConversationActivity conversationActivityOrientation2 = conversationActivityController2.get();
+        assertNotNull(conversationActivityOrientation2);
+    }
+
+    @Test
+    @Ignore
+    public void testConversationStartedFromSDK() throws Exception {
+        SwrveTestUtils.loadCampaignsFromFile(mActivity, swrveSpy, "conversation_campaign.json", "8d4f969706e6bf2aa344d6690496ecfdefc89f1f");
+        SwrveConversation conversation = swrveSpy.getConversationForEvent("swrve.messages.showatsessionstart", new HashMap<String, String>());
+        assertNotNull(conversation);
+
+        Robolectric.flushForegroundThreadScheduler(); // allow tasks that added to ui thread to run (like activity.runOnUiThread)
+
+        assertNotNull(swrveSpy.campaigns);
+        // Campaign will be displayed on SDK init!
+        // Next activity started should be the ConversationActivity
+        ShadowActivity shadowMainActivity = Shadows.shadowOf(mActivity);
+        Intent nextIntent = shadowMainActivity.peekNextStartedActivity();
+        assertNotNull(nextIntent);
+        assertEquals(nextIntent.getComponent(), new ComponentName(mActivity, ConversationActivity.class));
+
+        // Impressions should have increased by 1
+        assertEquals(1, swrveSpy.campaigns.get(0).getImpressions());
+    }
+
+    @Test
+    public void testConversationWidths() throws Exception {
```

Comment:
This is the test for Conversation landscape:
https://swrvedev.jira.com/browse/SWRVE-12894